libr/util/astr.c

```diff
@@ -43,7 +43,11 @@ R_API RASN1String *r_asn1_concatenate_strings (RASN1String *s0, RASN1String *s1,
 		r_asn1_free_string (s0);
 		r_asn1_free_string (s1);
 	}
-	return r_asn1_create_string (str, true, len);
+	RASN1String* asn1str = r_asn1_create_string (str, true, len);
```

Comment:
this seems to be a pretty bad design api. the ownership of str shouldn't be transferred conditionally

cc @wargio

Reply:
RASN1String is meant for that.